app/src/main/java/net/sourceforge/opencamera/ui/RecSyncView.java

```diff
@@ -61,87 +74,71 @@ public RecSyncView(Context context) {
             Log.d(TAG, "small_screen: " + small_screen);
         }
 
-        final Preview preview = main_activity.getPreview();
+        button_sync_settings = addButton(R.string.sync_settings_unlocked, view -> {
+            if (MyDebug.LOG) {
+                Log.d(TAG, "clicked to buttonSyncSettings");
+            }
 
-        // TODO if (support recSync)
+            main_activity.clickedSyncSettings();
```

Comment:
view don't need to have activity ref. For listeners of clicks use setClickListener (custom) from activity. For some logic with SoftwareSyncController use activity or special extra class

Reply:
So you mean that we have to create a `setClickListener` method in MainActivity and use it to create listeners for buttons? But why don't just use button's `setOnClickListener` like we do now? And even if we create this method in MainActivity we still need to somehow obtain MainActivity in the view.

Reply:
setClickListener should be in this view class -> it will save function reference and when necessary call it

Reply:
The `addButton` method contains the call of button's `setOnClickListener` and provides the given listener to it. Do you mean that we need to create a separate function just to wrap a call to `setOnClickListener`? It's strange...

Reply:
No, you need to set this function from the outside -> in main activity: recSyncView.setSettingsClickListener ({
clickedSyncSettings();
}) and in view class just use provided lambda

Reply:
All of this is done to remove main activity from view as it creates strong dependency of the view to this activity